terraform/cluster/aws/main.tf

```diff
@@ -263,6 +263,21 @@ resource "local_file" "kubeconfig" {
   })
 }
 
+resource "null_resource" "user_data_download" {
+  count = var.user_data_url != "" ? 1 : 0 
+
+  provisioner "local-exec" {
```

Comment:
please use https://registry.terraform.io/providers/hashicorp/http/latest/docs/data-sources/http

instead of local exec